### PR TITLE
Change the policy of dependencies version requirements

### PR DESCRIPTION
**Pull request by DeLion13**
In this pull request:

- Suggestion to replace `tilde`(`~`) requirement with `caret`(`^`) one to avoid a conflicts with other crates that requires a fresher version of the same crate in the same range of major release
- Update of some dependencies to the last versions if it was possible without interrupting into a code
- Removed semicolon that's prevented to warning: `warning: trailing semicolon in macro used in expression position` (while building locally on `Ubuntu 20.10`)

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -13,20 +13,20 @@ readme = "README.md"
 license = "MIT"
 
 [dev-dependencies]
-nix = "~0.11"
-tempdir = "~0.3"
+nix = "^0.23"
```
Comment (Veetaha) [inline Cargo.toml]: Caret requirement is the [default requirement in cargo](https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#specifying-dependencies-from-cratesio). So we may make it simpler 
```suggestion
nix = "0.23"
```

**Review by Veetaha — APPROVED** [review]

**Comment by cholcombe973** [comment]
Looks good! Thanks for doing this

**Review by cholcombe973 — APPROVED** [review]

**Comment by cholcombe973** [comment]
Let's just make sure build works properly and I'll merge it

File: Cargo.toml (hunk 1 above)
Comment (DeLion13) [reply Cargo.toml]: Yes, we can remove that.
@cholcombe973 I corrected it to format `"x.y.z"` instead of `^x.y.z`

**Review by cholcombe973 — APPROVED** [review]

**Pull request merged**